addon/mixins/adapter-fetch.js

```diff
@@ -73,15 +73,20 @@ export function headersToObject(headers) {
  * Helper function that translates the options passed to `jQuery.ajax` into a format that `fetch` expects.
  * @param {Object} options
  */
-export function mungOptionsForFetch(_options) {
+export function mungOptionsForFetch(_options, adapter) {
   const options = (assign || merge)({
     credentials: 'same-origin',
   }, _options);
 
+  let adapterHeaders = adapter.get('headers');
+  if (adapterHeaders) {
+    options.headers = assign({}, options.headers || {}, adapterHeaders);
+  }
   options.method = options.type;
 
   // Mimics the default behavior in Ember Data's `ajaxOptions`
-  if (options.headers && (!options.headers['Content-Type'] || !options.headers['content-type'])) {
+  if (options.headers === undefined || !(options.headers['Content-Type'] || options.headers['content-type'])) {
```

Comment:
I think that that the logic was wrong. The tests I added explain it better.
The idea is that we only want to add the default content-type header if the user didn't provide their own. With the previous logic before, if the user provided a `'Content-Type'`, it was being overriden by the default because `!options.headers['content-type']` was true.